samplePDF/samplePDFFDBase.cpp

```diff
@@ -1414,11 +1358,103 @@ void samplePDFFDBase::addData(TH2D* Data) {
   }
 }
 
+void samplePDFFDBase::SetupNuOscillator() {
+  OscProbCalcerFactory* OscProbCalcFactory = new OscProbCalcerFactory();
+  OscillatorFactory* OscillFactory = new OscillatorFactory();
+  
+  NuOscProbCalcers = std::vector<OscillatorBase*>((int)MCSamples.size());
+  for (int iSample=0;iSample<(int)MCSamples.size();iSample++) {
+    std::cout << "Setting up NuOscillator::Oscillator object in OscillationChannel: " << iSample << "/" << MCSamples.size() << " ====================" << std::endl;
+    NuOscProbCalcers[iSample] = OscillFactory->CreateOscillator(NuOscillatorConfigFile);
+
+    std::vector<_float_> EnergyArray((int)MCSamples[iSample].nEvents);
+    for (int iEvent=0;iEvent<(int)MCSamples[iSample].nEvents;iEvent++) {
+      EnergyArray[iEvent] = *(MCSamples[iSample].rw_etru[iEvent]);
+    }
+    std::sort(EnergyArray.begin(),EnergyArray.end());
+
+    if (!NuOscProbCalcers[iSample]->EvalPointsSetInConstructor()) {
+	  std::cout << "Made it through EvalPointsSetInConstructor check" << std::endl;
+      NuOscProbCalcers[iSample]->SetEnergyArrayInCalcer(EnergyArray);
+
+      //============================================================================
+      //DB Atmospheric only part
+      if (MCSamples[iSample].rw_truecz != NULL) { //Can only happen if truecz has been initialised within the experiment specific code
+	std::vector<_float_> CosineZArray((int)MCSamples[iSample].nEvents);
+	for (int iEvent=0;iEvent<(int)MCSamples[iSample].nEvents;iEvent++) {
+	  CosineZArray[iEvent] = *(MCSamples[iSample].rw_truecz[iEvent]);
+	}
+	std::sort(CosineZArray.begin(),CosineZArray.end());
+	
+	NuOscProbCalcers[iSample]->SetCosineZArrayInCalcer(CosineZArray);
+      }
+      //============================================================================
+      
+    }
```

Comment:
As complete newbie should't here be check for NC events?
Otherwise you have larger vector and I htink NuOsc will calclatue for NC events. 
Weights will be correctly assgiend it just about SPEEEED

Reply:
Yeah sure, I guess we could add check for NC events and remove from array - we're just lazy...

NC events don't get assigned any probabilities: https://github.com/mach3-software/MaCh3/blob/5fa87ded750b3ba2f040e7eddd9e51e7dcc7350b/samplePDF/samplePDFFDBase.cpp#L1399-L1404

So yes, we should make an effort to remove NC event energies/cosines from the array

Reply:
NC events now removed from the array which is passed to the NuOscillator objects